packages/astro/src/assets/utils/node/emitAsset.ts

```diff
@@ -62,14 +79,34 @@ export async function emitESMImage(
 		const filename = path.basename(pathname, path.extname(pathname) + `.${fileMetadata.format}`);
 
 		try {
-			// fileEmitter throws in dev
-			const handle = fileEmitter!({
-				name: filename,
-				source: await fs.readFile(url),
-				type: 'asset',
-			});
-
-			emittedImage.src = `__ASTRO_ASSET_IMAGE__${handle}__`;
+			// SVG deduplication: check if this content already exists
+			if (fileMetadata.format === 'svg') {
```

Comment:
There's a bit of repeated code here. Before, it was a few lines, so it was easy to maintain. However, it's now quite long; we should move the code inside a function.

Reply:
Sorry, I made a mistake